pages/studentPortal/web/React/assignments/template.tsx

```diff
@@ -11,44 +11,72 @@ import { Button } from "../../../../../components/ui/Button";
 import {
   FETCH_USER_ASSIGNMENT_SUBMISSIONS,
   FetchUserAssignmentSubmissionsDataResponse,
-  UserAssignmentSubmissionsData,
 } from "../../../../../graphql/fetchUserAssignmentSubmissions";
 import { useAuth } from "../../../../../lib/authContext";
+import {
+  assignmentsSelector,
+  setUserAssignments,
+} from "../../../../../redux/assignmentsSlice";
+import { selectionSetMatchesResult } from "@apollo/client/cache/inmemory/helpers";
 
 const React2 = ({ incompleteStage, submittedStage, completedStage }) => {
   const router = useRouter();
   const [stage, setStage] = useState(0);
   const { user } = useAuth();
-  const [assignments, setAssignments] = useState<
-    UserAssignmentSubmissionsData[]
-  >([]);
+  const { userAssignments } = useSelector(assignmentsSelector);
+  const dispatch = useDispatch();
+
   // REQUIRED: create assignment in coding_assignments table to generate ID and paste here
   const assignmentId = "2cf9156a-4f6f-452d-b09a-2c54f19a7b40";
 
-  const { data } = useQuery<FetchUserAssignmentSubmissionsDataResponse>(
-    FETCH_USER_ASSIGNMENT_SUBMISSIONS,
-    {
-      variables: {
-        user_id: user.uid,
-        assignmentId: assignmentId,
-      },
+  useEffect(() => {
+    if (userAssignments) {
+      const currentAssignment = userAssignments.filter(
+        (assignment) => assignment.assignmentId === assignmentId
+      );
+      let assignmentReviewLink = currentAssignment.map(
+        (assignment) => assignment.review_link
+      );
+      let assignmentSubmissionLink = currentAssignment.map(
+        (assignment) => assignment.submission_link
+      );
+      if (assignmentReviewLink != null) {
+        setStage(Stage.COMPLETED);
+      } else if (assignmentSubmissionLink.length > 0) {
+        setStage(Stage.SUBMITTED);
+      } else {
+        setStage(Stage.INCOMPLETE);
+      }
+    } else {
+      fetchUserAssignmentSubmissions();
+    }
+  }, [userAssignments]);
 
-      onCompleted: (data: FetchUserAssignmentSubmissionsDataResponse) => {
-        if (data.user_assignment_submissions.length > 0) {
-          setAssignments(data.user_assignment_submissions);
-          if (data.user_assignment_submissions[0].review_link != null) {
-            setStage(Stage.COMPLETED);
-          } else if (
-            data.user_assignment_submissions[0].submission_link.length > 0
-          ) {
-            setStage(Stage.SUBMITTED);
-          } else {
-            setStage(Stage.INCOMPLETE);
+  const [fetchUserAssignmentSubmissions] =
+    useLazyQuery<FetchUserAssignmentSubmissionsDataResponse>(
+      FETCH_USER_ASSIGNMENT_SUBMISSIONS,
+      {
+        variables: {
+          user_id: user.uid,
+          assignmentId: assignmentId,
+        },
+
+        onCompleted: (data: FetchUserAssignmentSubmissionsDataResponse) => {
+          if (data.user_assignment_submissions.length > 0) {
```

Comment:
i think the logic for checking the DB response and the redux state can be handled by a singular function so you don't have to update logic in both places

you'll just have to write the function so it can work in either case